pkg/nodeconfig/nodeconfig.go

```diff
@@ -265,6 +289,55 @@ func (nc *nodeConfig) applyLabelsAndAnnotations(labels, annotations map[string]s
 	return nil
 }
 
+// isCloudControllerOwnedByCCM checks if Cloud Controllers are managed by Cloud Controller Manager (CCM)
+// instead of Kube Controller Manager.
+func isCloudControllerOwnedByCCM() (bool, error) {
+	cfg, err := crclientcfg.GetConfig()
+	if err != nil {
+		return false, errors.Wrap(err, "unable to get config to talk to kubernetes api server")
+	}
+
+	client, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return false, errors.Wrap(err, "unable to get client from the given config")
+	}
+
+	return cluster.IsCloudControllerOwnedByCCM(client)
+}
+
+// ensureCloudTaintIsNotSet makes sure that the taint doesn't exist on a fully configured node
+func (nc *nodeConfig) ensureCloudTaintIsNotSet() error {
```

Comment:
It looks like this function be replaced with [RemoveTaint()](https://pkg.go.dev/k8s.io/kubernetes/pkg/util/taints#RemoveTaint)?

> RemoveTaint tries to remove a taint from annotations list. Returns a new copy of updated Node and true if something was updated false otherwise.